master/getting-started/kubernetes/tutorials/app-layer-policy/index.md

```diff
@@ -0,0 +1,298 @@
+
+ * All incoming connections from workloads in the Istio mesh are equally trusted
+ * Possession of a key & certificate pair is the *only* access credential considered.
+ 
+To understand why these might be a problem, let's take them one at a time.
+
+#### Trusting workloads
+
+Trusting connections from any workload in the Istio mesh is a poor security architecture because, 
+like Kubernetes, Istio is designed to host multiple applications.  Some of those applications may 
+not be as trusted as others.  They may be operated by different users or teams with wildly different
+security requirements.  We don't want our secure financial application microservices accessible from
+some hacky prototype another developer is cooking up.
+
+Even within our own application, the best practice is to limit access as much as possible.  Only 
+pods that need access to a service should get it.  Consider the YAO Bank application.  The customer
+web service does not need, and should not have direct access to the backend database.  The customer 
+web service needs to directly interact with clients outside the cluster, some of whom may be
+malicious.  Unfortunately, vulnerabilities in web applications are all too common.  For example, an
+[unpatched vulnerabiltiy in Apache Struts][struts cve] is what allowed attackers their initial
+access into the Equifax network where they then launched a devastating attack to steal millions of
+people's financial information.
+
+Imagine what would happen if an attacker were to gain control of the customer web pod in our
+application.
+
+Let's simulate this by `exec`'ing into that pod.
+
+    kubectl exec -ti customer-<fill in pod ID> -c customer bash
+
+You should get a bash shell inside the customer pod.  Notice that from here, we get direct access to
+the backend database.  For example, we can list all the entries in the database like this:
+
+    curl http://database:2379/v2/keys?recursive=true | python -m json.tool
+
+(Piping to `python -m json.tool` nicely formats the output.)
+
+#### Single factor authorization
+
+The possession of a key and certificate pair is a very strong assertion that a connection is 
+authentic because it is based on cryptographic proofs that are believed to be nearly impossible to
+forge.  When we authenticate connections this way we can say with extremely high confidence that the
+party on the other end is in possession of the corresponding key. However, this is only a proxy for
+what we actually want to be confident of: that the party on the other end really is the authorized
+workload we want to communicate with.  Keeping the private key a secret is vital to this confidence,
+and occasionally attackers can find ways to trick applications into giving up secrets they should
+not.  For example, the [Heartbleed] vulnerability in OpenSSL allowed attackers to trick an affected
+application into reading out portions of its memory, compromising private keys (among other
+confidential information).
```

Comment:
Personally, I think the parentheses here are unnecessary and could profitably be deleted.